### PR TITLE
fix(components): [tree-v2] modify the condition for when the value of currentKey is 0

### LINKED ISSUE #13933: [Component] [tree-v2] treeV2组件 key为0时  is-current 这个类名没有赋给节点元素   is-current的判断能否排除Number 0的情况
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.4`
- Element Plus Version: `2.3.9`
- Browser / OS: `macos`
- Build Tool: `Vite`

### Related Component

- `el-tree-v2`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHZlcnNpb24gYXMgdnVlVmVyc2lvbiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHZlcnNpb24gYXMgZXBWZXJzaW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgRWxlbWVudFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcblxuY29uc3QgbXNnID0gcmVmKCdIZWxsbyBXb3JsZCEnKVxuXG5jb25zdCBkYXRhPXJlZihbXG4gIHtcbiAgICBpZDowLFxuICAgIGxhYmVsOicxMTExMSdcbiAgfSxcbiAge1xuICAgIGlkOjEsXG4gICAgbGFiZWw6JzIyMjIyJ1xuICB9LFxuICB7XG4gICAgaWQ6MixcbiAgICBsYWJlbDonMzMzMzMnXG4gIH1cbl0pXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuXG48ZWwtdHJlZS12MiAgIDpkYXRhPVwiZGF0YVwiPlxuPC9lbC10cmVlLXYyPlxuPC90ZW1wbGF0ZT5cblxuPHN0eWxlPlxuICAgLmVsLXRyZWUtbm9kZS5pcy1jdXJyZW50IHtcbiAgICAgIC5lbC10cmVlLW5vZGVfX2NvbnRlbnQge1xuICAgICAgICAgIGJhY2tncm91bmQtY29sb3I6ICMwMDgxZmY7XG4gICAgICB9XG4gICAgfVxuPC9zdHlsZT4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl1cbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

### Steps to reproduce

点击节点，当节点key为0时，setCurrentKey这个方法无效。is-current这个类名并没有正确赋予元素。

### What is Expected?

setCurrentKey 方法起作用，或者is-current这个类名能正常赋予元素

### What is actually happening?

setCurrentKey不起作用  is-current类名未赋予元素  

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by yj-liuzepeng**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

## Related Issue

Fixes #13933

### CODE REVIEW

**Review by btea — APPROVED** [review]

**Review by ryuhangyeong — APPROVED** [review]

**Comment by btea** [comment]
@yj-liuzepeng Could you add a test case?

**Comment by yj-liuzepeng** [comment]
> 



> @yj-liuzepeng Could you add a test case?

I'm not familiar with unit testing, but I added a test case. Can you review it, please? @btea

File: packages/components/tree-v2/__tests__/tree.test.ts [hunk 1]
```diff
@@ -1176,5 +1176,65 @@ describe('Virtual Tree', () => {
...
       expect(treeRef.getCurrentKey()).toBe('1-1-2')
     })
+
+    test('setCurrentKey', async () => {
+      const { treeRef, wrapper } = createTree({
+        data() {
+          return {
+            data: [
+              {
+                id: '0',
```
Comment (btea) [inline packages/components/tree-v2/__tests__/tree.test.ts:1186]:
```suggestion
                id: 0,
```

File: packages/components/tree-v2/__tests__/tree.test.ts [hunk 2]
```diff
@@ -1176,5 +1176,65 @@ describe('Virtual Tree', () => {
...
+              {
+                id: '2',
+                label: 'node-2',
+              },
+            ],
+          }
+        },
+      })
+      await nextTick()
+      treeRef.setCurrentKey('0')
```
Comment (btea) [inline packages/components/tree-v2/__tests__/tree.test.ts:1232]:
```suggestion
      treeRef.setCurrentKey(0)
```

File: packages/components/tree-v2/__tests__/tree.test.ts [hunk 3]
```diff
@@ -1176,5 +1176,65 @@ describe('Virtual Tree', () => {
...
+      await nextTick()
+      const currentKeys = treeRef.getCurrentKey()
+      const nodes = wrapper.findAll(TREE_NODE_CLASS_NAME)
+      expect(currentKeys.toString()).toBe(['0'].toString())
```
Comment (btea) [inline packages/components/tree-v2/__tests__/tree.test.ts:1236]:
```suggestion
      expect(currentKeys).toBe(0)
```

**Pull request merged**